pkg/keccak/contrib/Modes/keccak1600.h

```diff
@@ -0,0 +1,169 @@
+http://ketje.noekeon.org/
+
+To the extent possible under law, the implementer has waived all copyright
+and related or neighboring rights to the source code in this file.
+http://creativecommons.org/publicdomain/zero/1.0/
+
+This is free and unencumbered software released into the public domain.
+
+Anyone is free to copy, modify, publish, use, compile, sell, or
+distribute this software, either in source code form or as a compiled
+binary, for any purpose, commercial or non-commercial, and by any
+means.
+
+In jurisdictions that recognize copyright laws, the author or authors
+of this software dedicate any and all copyright interest in the
+software to the public domain. We make this dedication for the benefit
+of the public at large and to the detriment of our heirs and
+successors. We intend this dedication to be an overt act of
+relinquishment in perpetuity of all present and future rights to this
+software under copyright law.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+OTHER DEALINGS IN THE SOFTWARE.
+
+For more information, please refer to <http://unlicense.org/>
+*/
+
+#ifndef KECCAK1600_H
+#define KECCAK1600_H
+
+#define KeccakP200_excluded
+#define KeccakP400_excluded
+
+#ifndef KeccakP1600_excluded
+
+#include "keccak_defs.h"
+#include "KeccakHash.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define keccak1600hash_instance Keccak_HashInstance
+#define fixed_output_length     fixedOutputLength
+#define delimited_suffix        delimitedSuffix
+
+/**
+  * Function to initialize the Keccak[r, c] sponge function instance used in sequential hashing mode.
+  * @param  hash_instance    Pointer to the hash instance to be initialized.
+  * @param  rate        The value of the rate r.
+  * @param  capacity    The value of the capacity c.
+  * @param  hashbitlen  The desired number of output bits,
+  *                     or 0 for an arbitrarily-long output.
+  * @param  delimited_suffix Bits that will be automatically appended to the end
+  *                         of the input message, as in domain separation.
+  *                         This is a byte containing from 0 to 7 bits
+  *                         formatted like the @a delimitedData parameter of
+  *                         the Keccak_SpongeAbsorbLastFewBits() function.
+  * @pre    One must have r+c=1600 and the rate a multiple of 8 bits in this implementation.
+  * @return SUCCESS if successful, FAIL otherwise.
+  */
+#define keccak1600hash_initialize(hash_instance, rate, capacity, hashbitlen, delimited_suffix)  Keccak_HashInitialize(hash_instance, rate, capacity, hashbitlen, delimited_suffix)
```

Comment:
Line-length (+ again doubting the necessity of this macro; if required, why not inline function as with the SHA3 functions?)

Reply:
Well, we're basically just giving an alias to a function here so I thought a macro was appropriate but I can change it to a inline function if you think that's better. And again, this is necessary for a consistent naming scheme with keccak800.h.

Reply:
> Is there anything really speaking against the usage of a macro in this case?

1. coding conventions, thus consistency throughout the code base
2. the idea is that if something *can* be expressed in pure C (without *severly* reduced readability), we do it in C. That way, all tooling, debugging, ... is made easier.

Reply:
Alright, this will be pushed along with all other changes then.